common/src/View/MapDocumentCommandFacade.cpp

```diff
@@ -424,53 +428,24 @@ namespace TrenchBroom {
             groupWasClosedNotifier(previousGroup);
         }
 
-        class CanTransformVisitor : public Model::ConstNodeVisitor, public Model::NodeQuery<bool> {
-        private:
-            vm::mat4x4 m_transform;
-            vm::bbox3 m_worldBounds;
-        public:
-            CanTransformVisitor(const vm::mat4x4& transform, const vm::bbox3& worldBounds) :
-                m_transform(transform),
-                m_worldBounds(worldBounds) {}
-        private:
-            void doVisit(const Model::WorldNode*) override  { setResult(true); }
-            void doVisit(const Model::LayerNode*) override  { setResult(true); }
-            void doVisit(const Model::GroupNode*) override  { setResult(true); }
-            void doVisit(const Model::EntityNode*) override { setResult(true); }
-            void doVisit(const Model::BrushNode* brushNode) override {
-                const Model::Brush& brush = brushNode->brush();
-                setResult(brush.canTransform(m_transform, m_worldBounds));
-            }
-            bool doCombineResults(const bool oldResult, const bool newResult) const override {
-                return newResult && oldResult;
-            }
-        };
-
         bool MapDocumentCommandFacade::performTransform(const vm::mat4x4 &transform, const bool lockTextures) {
-          // Test whether all brushes can be transformed; abort if any fail.
```

Comment:
If I understand right, this was removed because the `TransformObjectsCommand::doPerformDo` now restores the snapshot if the "perform" fails?

Reply:
Also might be worth documenting what it means if `MapDocumentCommandFacade::performTransform` returns false, namely some of the nodes may have been transformed and it's up to the caller to clean up everything.

Reply:
Updated the existing comment in 051e73067.